### PR TITLE
Pages on contributing to MDAnalysis

### PR DESCRIPTION
**Pull request by lilyminium**
Many new contributors will not be aware of the available pages on contributing to MDAnalysis, as they aren't indexed anywhere (just one of the list of 100+ pages). Many of the wiki pages on contributing to MDAnalysis also cover redundant or outdated information, or spread it out in a way that makes it hard to get the whole picture. I've collected most of the information into the user guide, focusing more on the more recently updated pages:

- [Contributing to MDAnalysis](https://lilyminium.github.io/UserGuide/contributing.html) (the main page)
- [Preparing releases and hotfixes](https://lilyminium.github.io/UserGuide/preparing_releases_and_hotfixes.html)
- [Module imports in MDAnalysis](https://lilyminium.github.io/UserGuide/module_imports.html)
- [Tests in MDAnalysis](https://lilyminium.github.io/UserGuide/testing.html)

Pages incorporated into the user guide:

- [x] [AppVeyor](https://github.com/MDAnalysis/mdanalysis/wiki/AppVeyor) (last edited April 2019) 
- [x] [Contributing to MDAnalysis.analysis](https://github.com/MDAnalysis/mdanalysis/wiki/Contributing-to-MDAnalysis.analysis) (last edited 2015)
    - Rules about [maintaining code](https://github.com/MDAnalysis/mdanalysis/wiki/Contributing-to-MDAnalysis.analysis#standards-for-code-in-mdanalysisanalysis) have not been added. Firstly, ``git blame`` tells you who contributed the code. More importantly, this is a huge responsibility that would make many think again about contributing code. If this is important, please let me know!
- [x] [ContributingCode](https://github.com/MDAnalysis/mdanalysis/wiki/ContributingCode) (last edited 2017)
- [x] [CoreDevelopment](https://github.com/MDAnalysis/mdanalysis/wiki/CoreDevelopment) (last edited 2017)
- [x] [DevelopmentBranch](https://github.com/MDAnalysis/mdanalysis/wiki/DevelopmentBranch) (last edited August 2019)
- [x] [DevelopmentWorkflow](https://github.com/MDAnalysis/mdanalysis/wiki/DevelopmentWorkflow) (last edited 2015)
- [x] [DistributedDevelopment](https://github.com/MDAnalysis/mdanalysis/wiki/DistributedDevelopment) (last edited 2017)
- ~[Documentation Standards](https://github.com/MDAnalysis/mdanalysis/wiki/Documentation-Standards)~ (last edited 2015) seemed moot
- [x] [PreparingReleases](https://github.com/MDAnalysis/mdanalysis/wiki/PreparingReleases) (last edited 2018)
    - Hopefully I mashed this with DevelopmentWorkflow correctly
- [x] [RepositoryStructure](https://github.com/MDAnalysis/mdanalysis/wiki/RepositoryStructure) (last edited 2015)
- [x] [SemanticVersioning](https://github.com/MDAnalysis/mdanalysis/wiki/SemanticVersioning) (last edited 2015)
- [x] [Setup Development Environment](https://github.com/MDAnalysis/mdanalysis/wiki/Setup-Development-Environment) (last edited 2017)
- [x] [Source](https://github.com/MDAnalysis/mdanalysis/wiki/Source) (last edited 2015)
- [x] [Style guide](https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide) (last edited 2017)
- [x] [Style Guide: git commits](https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide:-git-commits) (last edited 2015)
- [x] [Style Guide: Tests](https://github.com/MDAnalysis/mdanalysis/wiki/Style-Guide:-Tests) (last edited 2017)
- [x] [UnitTests](https://github.com/MDAnalysis/mdanalysis/wiki/UnitTests) (last edited July 2019)
- [x] [WritingDocumentation](https://github.com/MDAnalysis/mdanalysis/wiki/WritingDocumentation#location-of-documentation) (last edited April 2019)

I have no experience with `virtualenv`, so I'm not certain that section is as helpful as it could be. I similarly hope I noted the workflow for preparing releases correctly.

I also have never used zsh, which I understand will become (is?) the default terminal shell on the new Macs.

Finally, if MDAnalysis has a collection of issues with good-first-issue flairs, that would be very helpful to link.

### CODE REVIEW

File: doc/source/contributing.rst [hunk 1]
```diff
@@ -5,22 +5,853 @@
...
-.. _`Code of Conduct`: https://www.mdanalysis.org/pages/conduct/
+.. note::
+
+    Parts of this page came from the `Contributing to pandas <http://pandas.pydata.org/pandas-docs/stable/contributing.html>`_ guide.
+
+Where to start?
+===============
+
+All contributions, bug reports, bug fixes, documentation improvements,
+enhancements, and ideas are welcome.
```
Comment (richardjgowers) [inline doc/source/contributing.rst]: I think somewhere in this page there needs to be a quick paragraph explaining the license used (GPL) and that contributions made will need to fall under this.

File: doc/source/preparing_releases_and_hotfixes.rst [hunk 2]
```diff
@@ -0,0 +1,206 @@
+.. -*- coding: utf-8 -*-
```
Comment (richardjgowers) [inline doc/source/preparing_releases_and_hotfixes.rst]: Speaking as the release manager, I don't think we do hotfixes?  I think I'd sooner just push out a new release and call the old one broken.

File: doc/source/module_imports.rst [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+List of core module dependencies
+--------------------------------
+
+Any module from the standard library can be used, as well as the following nonstandard libraries:
+
+   * :mod:`six`
+   * :mod:`numpy`
+   * :mod:`biopython`
+   * :mod:`gridDataFormats`
+   * :mod:`networkx`
```
Comment (richardjgowers) [inline doc/source/module_imports.rst]: networkx and joblib are only analysis/ submodules

**Review by richardjgowers — COMMENTED** [review]
these look good @lilyminium .  I pushed some minor fixes and made a few comments

File: doc/source/preparing_releases_and_hotfixes.rst (hunk 2 above)
Comment (lilyminium) [reply doc/source/preparing_releases_and_hotfixes.rst]: Fair enough --[ removed that section](https://lilyminium.github.io/UserGuide/preparing_releases_and_hotfixes.html).

File: doc/source/contributing.rst (hunk 1 above)
Comment (lilyminium) [reply doc/source/contributing.rst]: Thanks for the review! What do you think of [this](https://lilyminium.github.io/UserGuide/contributing.html)?

**Comment by richardjgowers** [comment]
Yep looks really good

**Pull request merged**